src/acl/Gadgets.cc

```diff
@@ -161,20 +165,23 @@ aclParseAccessLine(const char *directive, ConfigParser &, acl_access **treep)
 
     /* Append to the end of this list */
 
+    assert(config);
+    if (!*config)
+        *config = new acl_access();
+    const auto treep = *config;
+
     assert(treep);
```

Comment:
It looks like this assertion is unnecessary - we just checked and assigned the pointer two lines above.

Reply:
Agreed.

Rationale: This high-risk PR goes an extra mile to preserve original code to the extent possible. This is why I left that original (and now very paranoid!) assertion intact and preserved the original treep-setting code rather then merging it with PR-added `new acl_access` code above.

You may notice another sign of this approach in the nearby code: The earlier `config && *config` condition does not account for the asserted fact that `config` is not nil. Again, I am preserving original code structure to clarify (the scope of) PR changes and reduce "Are you sure this is sane/safe?" concerns.

Ideally, this PR should be followed up with a PR that improves this code while not being bound by this PR code preservation and diff clarity constraints.